homeassistant/components/growatt_server/coordinator.py

```diff
@@ -0,0 +1,215 @@
+"""Coordinator module for managing Growatt data fetching."""
+
+import datetime
+import json
+import logging
+from typing import Any
+
+import growattServer
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import dt as dt_util
+
+from .const import DEFAULT_URL, DOMAIN
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GrowattCoordinator(DataUpdateCoordinator):
```

Comment:
```suggestion
class GrowattCoordinator(DataUpdateCoordinator[..]):
```
Can we type the return type of `_async_update_data`?